src/condor_tools/history.cpp

```diff
@@ -497,6 +496,31 @@ main(int argc, const char* argv[])
 			if ( *pcolon == 'd' || *pcolon == 'D' ) { delete_epoch_ads = true; break; }
 		}
 	}
+	else if (is_dash_arg_prefix(argv[i], "type", 1)) { // Purposefully undocumented (Intended internal use)
+		if (argc <= i+1 || *(argv[i+1]) == '-') {
+			fprintf(stderr, "Error: Argument %s requires another parameter\n", argv[i]);
+			exit(1);
+		}
+		++i; // Claim the next argument for this option even if we don't use it
+		// Process passed Filter Ad Types if 'ALL' has not been specified
+		if ( ! filterAdTypes.contains("ALL")) {
+			StringTokenIterator adTypes(argv[i]);
+			for (auto type : adTypes) {
+				upper_case(type);
+				if (type == "ALL") {
+					// 'ALL' will not filter out any ad types so add to set and break
```

Comment:
This still leaves things other than ALL in filterAdTypes, potentially.  Is that OK?

Reply:
I added this comment early -- the remaining code handles it just fine.  Technically a little less efficient than it could be, but not in a way that should ever matter.